### PR TITLE
Support `captureFeedback`

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Implementation based on https://develop.sentry.dev/application/feedback-architecture/#feedback-events

## :bulb: Motivation and Context

Relates to #1593

## :green_heart: How did you test it?

Unit tests

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- [x] How can i test if the sent envelope is correct/expected on the server side?
  - [x] Envelope sent through native sdks
    - [x] iOS
    - [x] macOS
    - [ ] Android
  - [x] Envelope sent through http

### CODE REVIEW

File: dart/test/mocks/mock_transport.dart
Comment (denrase) [inline dart/test/mocks/mock_transport.dart:20]: Euclid's first common notion is this: "Things which are equal to the same things are equal to each other" ^^

**Comment by ueman** [comment]
I'm really looking forward to this, to use it in https://pub.dev/packages/feedback Thanks for tackling it!

**Comment by buenaflor** [comment]
@ueman  on a side note, after this PR we are also looking into implementing the [feedback widget](https://docs.sentry.io/platforms/javascript/user-feedback/#user-feedback-widget)

maybe it makes sense to adopt some code from your package? but let's see I haven't taken a deep dive yet into that

**Comment by denrase** [comment]
@buenaflor I'm not sure if i have all the pieces in place that we need, would you mind taking a quick look over it and giving me feedback? /cc ueman

**Comment by ueman** [comment]
> @ueman on a side note, after this PR we are also looking into implementing the [feedback widget](https://docs.sentry.io/platforms/javascript/user-feedback/#user-feedback-widget)
> 
> maybe it makes sense to adopt some code from your package? but let's see I haven't taken a deep dive yet into that

I would advise against using my package, at least as of know, as there are a couple of fundamental issues with it, that would require a lot of effort to fix. I would love to see those fixed though.

As for the (Sentry) feedback widget, there's already a very simple but working implementation at https://github.com/getsentry/sentry-dart/blob/main/flutter/example/lib/user_feedback_dialog.dart

With a bit of polishing (mainly around handling issues during sending the feedback, maybe UI adjustments), it should be quite easy to ship.

> @buenaflor I'm not sure if i have all the pieces in place that we need, would you mind taking a quick look over it and giving me feedback? /cc ueman

Unfortunately, I also already failed once to implement this spec. So I'm afraid I can't really help here :/

**Comment by denrase** [comment]
Ok, i can see the new feedback on sentry.io \o/

Bildschirmfoto 2024-08-13 um 16 45 21

**Comment by denrase** [comment]
Seems to work fine on iOS, macOS and web. On Android i can see the envelope in Spotlight, but no new issue is appearing on sentry.io.

Bildschirmfoto 2024-08-13 um 17 32 01

**Comment by denrase** [comment]
@buenaflor Could not figure out what the issue with Android is, other envelopes (errors/messages) get delivered.

**Comment by buenaflor** [comment]
@denrase any signs of the event being rejected for some reason in logcat?

**Comment by denrase** [comment]
@buenaflor I'm investigating the envelopes and there is not really a difference. Think we need someone from the backend team to help us out.

[event_android.json](https://github.com/user-attachments/files/16689913/event_android.json)
[event_ios.json](https://github.com/user-attachments/files/16689914/event_ios.json)

**Comment by buenaflor** [comment]
are there any logs being capture by android or is it successfully sending it to sentry but sentry is just not processing it?

**Comment by buenaflor** [comment]
I can confirm the envelope is being sent, could be a backend issue (or maybe it's a frontend issue)

**Review by buenaflor — COMMENTED** [review]
overall looks good, just need to check why android is not being displayed

let's wait for some resolution on this before merging this

**Comment by buenaflor** [comment]
according to @aliu39 this [issue](https://github.com/getsentry/sentry/issues/76094) might be relevant, let's wait for it to be fixed and see if android is still not working

**Comment by aliu39** [comment]
@buenaflor Can you test again to see if it's showing up?

About the other issue - the search backend we rely on has a bug where a feedback will be dropped from the response. This happens very rarely, and when it does it's still returned by other endpoints, ex count. Have you noticed your inbox counts not matching the number of feedbacks in the list?

**Comment by buenaflor** [comment]
cc @denrase can you try it out pls

**Comment by denrase** [comment]
@aliu39 @buenaflor Still not seeing Feedback from Android. Web works, so the client should be doing everything as expected.

Bildschirmfoto 2024-09-05 um 10 16 30

**Comment by buenaflor** [comment]
@aliu39 we believe its most likely an SDK internal issue, we'll investigate it

**Comment by denrase** [comment]
@buenaflor Waiting for https://github.com/getsentry/sentry-java/pull/3687

**Comment by denrase** [comment]
@buenaflor https://github.com/getsentry/sentry-java/pull/3687 was merged. Waiting for it beeing released, then we can finish here.

**Comment by ueman** [comment]
Can't this be already merged? I mean the code works, and the Android dependency will be updated anyway as soon as it's release right?

**Comment by denrase** [comment]
@ueman Think we could, but then we'd potentially block releases if Android takes longer than expected, no?

**Comment by denrase** [comment]
@buenaflor What do you think? Should we merge this with the info that Android support will come later? Would be easier, as i have to merge in main every other day and fixing issues that arise.

**Comment by buenaflor** [comment]
@denrase I'll do a release soon, either today or tomorrow, we can merge it afterwards so we can wait for the android release

**Review by buenaflor — APPROVED** [review]

**Comment by buenaflor** [comment]
just tested and android works fine after bumping to Android `7.15.0`👍 

will merge after tests are green

**Pull request merged**
